deployments/helm/nvidia-dra-driver-gpu/templates/_helpers.tpl

```diff
@@ -43,36 +43,47 @@ Create chart name and version as used by the chart label.
 {{- end }}
 
 {{/*
-Common labels
+Standard labels: documented at
+https://helm.sh/docs/chart_best_practices/labels/
+Apply this to all high-level objects (Deployment, DaemonSet, ...).
+Pod template labels are included here to deliver name+instance.
 */}}
 {{- define "nvidia-dra-driver-gpu.labels" -}}
 helm.sh/chart: {{ include "nvidia-dra-driver-gpu.chart" . }}
-{{ include "nvidia-dra-driver-gpu.templateLabels" . }}
-{{- if .Chart.AppVersion }}
 app.kubernetes.io/version: {{ .Chart.AppVersion | quote }}
-{{- end }}
 app.kubernetes.io/managed-by: {{ .Release.Service }}
+{{ include "nvidia-dra-driver-gpu.templateLabels" . }}
 {{- end }}
 
 {{/*
-Template labels
+Apply this to all pod templates (a smaller set of labels compared to
+the set of standard labels above, to not clutter individual pods too
+much). Note that these labels cannot be used to distinguish
+components within this Helm chart.
 */}}
 {{- define "nvidia-dra-driver-gpu.templateLabels" -}}
 app.kubernetes.io/name: {{ include "nvidia-dra-driver-gpu.name" . }}
 app.kubernetes.io/instance: {{ .Release.Name }}
-{{- if .Values.selectorLabelsOverride }}
-{{ toYaml .Values.selectorLabelsOverride }}
-{{- end }}
```

Comment:
Let's keep the selector label (single label, I propose) strictly independent.

"Template labels" always meant "pod template labels" and I think it really always meant "apply a common set of labels to all pods of this Helm chart".

By that nature, this set of labels cannot distinguish individual groups of pods within the Helm chart.

Which is what "selector labels" want to do.

In that sense, those two concepts (template labels and selector labels) are generally incompatible (which goes unnoticed for the special case of there being just one component in the Helm chart).